src/benchmark.cc

```diff
@@ -877,14 +882,17 @@ void ParseCommandLineFlags(int* argc, const char** argv) {
                        &FLAGS_benchmark_repetitions) ||
         ParseStringFlag(argv[i], "benchmark_format",
                         &FLAGS_benchmark_format) ||
-        ParseBoolFlag(argv[i], "color_print",
-                       &FLAGS_color_print) ||
+        ParseStringFlag(argv[i], "benchmark_color",
+                        &FLAGS_benchmark_color) ||
+        // Legacy flag for benchmark_color.
+        ParseStringFlag(argv[i], "color_print",
+                        &FLAGS_benchmark_color) ||
         ParseInt32Flag(argv[i], "v", &FLAGS_v)) {
       for (int j = i; j != *argc; ++j) argv[j] = argv[j + 1];
 
       --(*argc);
       --i;
-    } else if (IsFlag(argv[i], "help")) {
+    } else if (IsFlag(argv[i], "help") || HasBenchmarkFlagPrefix(argv[i])) {
```

Comment:
Not quite -- without this check, you could pass `--benchmark_format` or similar with no associated value, and no error will be reported.